### ISSUE TITLE
Getting exception on parallel run with new version (6.2.0)

### ISSUE THREAD

**Issue by himansja** [postmanlabs/newman#3269]
1. Newman Version (6.2.0`):
 2. OS details (type, version, and architecture):
 3. Are you using Newman as a library
 3. Did you encounter this recently, or has this bug always been there: Recently with new version
 4. Expected behaviour: Should allow to run test in parallel
 5. Command / script used to run Newman:
 6. Sample collection, and auxiliary files (minus the sensitive details):
 7. Screenshots (if applicable):

<!--
Steps to reproduce the problem:

Getting exception as :
node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: read ECONNRESET
    at TLSWrap.onStreamRead (node:internal/stream_base_commons:217:20)
Emitted 'error' event on TLSSocket instance at:
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -4077,
  code: 'ECONNRESET',
  syscall: 'read'
}

**Comment by codenirvana** [comment]
@himansja can you share a scaled-down script where you are noticing this behaviour?

**Comment by himansja** [comment]
@codenirvana  
```js
// Runs Postman collection based on the provided folder list (each folder executed in parallel).
async.map(foldersList, function runCollectionFromFolder(collectionFolder, done) {
  htmlFileName = "report_" + path.parse(environmentFile).name + "_" +  path.parse(collectionFile).name.replaceAll(' ', '') + collectionFolder.replaceAll(' ', '') + ".html",
  xmlFileName =  "TestResults_" + path.parse(environmentFile).name + "_" + path.parse(collectionFile).name.replaceAll(' ', '') + collectionFolder.replaceAll(' ', '') + ".xml",
  newman.run({
    collection: postmanCollection,
    environment: path.join(__dirname, environmentFile),
	sslClientCertList: path.join(__dirname, certificateFile),
	folder: collectionFolder.trim(),
	
	reporters: ['cli','junit', 'htmlextra'],
	reporter: 
		{
			cli: {
				/*noSummary: true,
				noSuccessAssertions: true,
				noBanner: true,*/
				silent:true
			},
			htmlextra: {
				template: path.join(__dirname,'templates','htmlreqres.hbs'),
				export: path.join(outputDirForHtmlReport, htmlFileName)
			},
			junit: {
				 export: path.join(__dirname, xmlFileName)
			}
		}
	}, done)
	.on('beforeItem',function (err,args) { //https://github.com/postmanlabs/newman/wiki/Newman-Run-Events#beforeitem
		console.log('  Running "'+ args.item.name + '" test from '+ collectionFolder + ' folder');	
	});
},
function (err, results) {
    err && console.error(err);
	let throwError = false;
	let anyTestsExecuted = false;
    results.forEach(function (result) {
		//https://npmdoc.github.io/node-npmdoc-newman/build/apidoc.html#apidoc.element.newman.summary.attachTimingTrackers
		
		if (result.run.executions.length>0){
			anyTestsExecuted=true;
		}
        var failures = result.run.failures;
		if (failures.length>0){
			throwError=true
			console.error('\x1b[31m' + result.run.stats.assertions.failed + " assertions failed: \x1b[0m");
			failures.forEach(
				function (failure) {
					console.error('Request name: \x1b[91m ' + failure.parent.name + " / " + failure.source.name + '\" \x1b[0m');
					console.error(' Test \x1b[33m \'' + failure.error.test + '\' \x1b[0m failed with error: \x1b[31m \"' + failure.error.message + '\" \x1b[0m');
					console.error('\x1b[95m***\x1b[0m');
				})
		}
	});
	
	if (throwError) {
		console.error('\x1b[33m One or more tests has failed \x1b[93m (see above for details on what has failed, or the pipeline artifacts folder, where html reports are stored) \x1b[0m');
		process.exit(1);
	}
	else if (!anyTestsExecuted) {
		console.log("\x1b[33m No tests have been executed, based on provided criteria.. \x1b[0m");
		process.exit(1);
	}
	else {
		console.log("\x1b[92m All tests passed! \x1b[0m");
	}
});
```

**Comment by parthverma1** [comment]
Hi @himansja, we were able to replicate this issue and have fixed it with 6.2.1. 
Feel free to re-open this issue if the problem recurs.
